plugins/openai/tests/test_openai_compliance_annotator.py

```diff
@@ -0,0 +1,109 @@
+import pytest
+from typing import List
+
+from newhelm.annotators.openai_compliance_annotator import (
+    ComplianceAnnotation,
+    OpenAIComplianceAnnotator,
+)
+from newhelm.prompt import ChatMessage, ChatPrompt, TextPrompt
+from newhelm.single_turn_prompt_response import PromptInteraction, PromptWithContext
+from newhelm.sut import SUTCompletion, SUTResponse
+
+Interaction = ComplianceAnnotation.Interaction
+Completion = ComplianceAnnotation.Completion
+
+
+def _make_interaction(completions: List[str] = [""]) -> PromptInteraction:
+    sut_completions = [SUTCompletion(text=text) for text in completions]
+    return PromptInteraction(
+        prompt=PromptWithContext(prompt=TextPrompt(text="The prompt")),
+        response=SUTResponse(completions=sut_completions),
+    )
+
+
+def _make_annotation(compliance: str) -> ComplianceAnnotation:
+    return ComplianceAnnotation(
+        interactions=[Interaction(completions=[Completion(compliance=compliance)])]
+    )
+
+
+class MockSUT:
+    def __init__(self, responses: List):
+        self.response_stack = list(reversed(responses))
+        self.requests_received: List[str] = []
+
+    def evaluate(self, request: str) -> str:
+        self.requests_received.append(request)
+        return self.response_stack.pop()
+
+    def translate_text_prompt(self, prompt: TextPrompt) -> str:
+        return prompt.text
+
+    def translate_response(self, request: str, response: str) -> SUTResponse:
+        assert response is not None
+        return SUTResponse(completions=[SUTCompletion(text=response)])
+
+
+def test_default_decoder():
+    responses = ["1_full_compliance", "2_full_refusal", "3_partial_refusal"]
+    annotator = OpenAIComplianceAnnotator()
+    annotator.model = MockSUT(responses)
```

Comment:
Instead of monkey patching, can we pass MockSUT as the `sut` argument to OpenAIComplianceAnnotator? Typing might want you to inherit MockSUT from PromptResponseSUT, but that shouldn't require much extra work. You can always implement abstracts you don't care about with `NotImplementedError`.

Reply:
Oh yep, forgot to update the tests after adding the `sut` parameter to OpenAIComplianceAnnotator. Fixed.